command/command.go

```diff
@@ -75,6 +75,23 @@ type Command interface {
 	Wait() error
 }
 
+// FormattedError ...
+type FormattedError struct {
```

Comment:
I suggest moving the error-related code to a new file (errors.go).

Reply:
Can we rename this error to `ExitError` or `ExitStatusError` or `CommandFailedError`, we only use this error type, if the command was executed successfully, but returned a non-zero exit status.

Reply:
Can we create a constructor for this error type, which takes care os the actual error formatting, to separate this concept from the command execution logic a bit more:

`func NewCommandFailedError(exitStatus int, prinatbleCmd string, reason error/[]string) ...`

if the reason is empty we could add the placeholder text as a reason (`check the command's output for details`)

Reply:
Ok, added errors.go.